### PR TITLE
Removed NOT_FOUND for empty results.

### PR DESCRIPTION
**Pull request by harshjain2**
### Description
Fix for the issue due to which 404 was returned when no PITs were present.

### Issues Resolved
https://github.com/opensearch-project/OpenSearch/issues/5938

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff
- [ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: server/src/main/java/org/opensearch/action/search/DeletePitResponse.java [hunk 1]
```diff
@@ -57,7 +56,6 @@ public List<DeletePitInfo> getDeletePitResults() {
      */
     @Override
     public RestStatus status() {
-        if (deletePitResults.isEmpty()) return NOT_FOUND;
```
Comment (reta) [inline server/src/main/java/org/opensearch/action/search/DeletePitResponse.java:60]: So what would be the payload returned from the method? Empty list of PITs?

**Review by reta — CHANGES_REQUESTED** [review]
Please add the test cases

File: server/src/main/java/org/opensearch/action/search/DeletePitResponse.java (hunk 1 above)
Comment (harshjain2) [reply server/src/main/java/org/opensearch/action/search/DeletePitResponse.java:60]: The server returns 200 StatusCode with { "pits": [] } as the body when there are no PITs to delete.

**Comment by harshjain2** [comment]
@reta I have fixed integration tests. By [this](https://github.com/opensearch-project/OpenSearch/pull/6544#pullrequestreview-1325244781) comment, do you mean unit tests?

**Comment by reta** [comment]
> @reta I have fixed integration tests. By [this](https://github.com/opensearch-project/OpenSearch/pull/6544#pullrequestreview-1325244781) comment, do you mean unit tests?

The rest-api-spec is fine, thank you

**Comment by reta** [comment]
Also, please fix DCO check, it is failing

**Comment by harshjain2** [comment]
@reta I hope it is fine to add the signature in the last commit only.

**Comment by reta** [comment]
> @reta I hope it is fine to add the signature in the last commit only.
> 
> [IGNORE THIS MESSAGE]

Thanks, DCO is fine, please add CHANGELOG entry, thank you

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - Support for HTTP/2 (server-side) ([#3847](https://github.com/opensearch-project/OpenSearch/pull/3847))
 - Add getter for path field in NestedQueryBuilder ([#4636](https://github.com/opensearch-project/OpenSearch/pull/4636))
 - Allow mmap to use new JDK-19 preview APIs in Apache Lucene 9.4+ ([#5151](https://github.com/opensearch-project/OpenSearch/pull/5151))
+- Removed NOT_FOUND for empty results. ([#6544](https://github.com/opensearch-project/OpenSearch/pull/6544))
```
Comment (dreamer-89) [inline CHANGELOG.md:11]: If this needs to be backported to `2.x` branch, we can directly add this under `[Unrelease 2.x]` section.

**Comment by dreamer-89** [comment]
> ### Gradle Check (Jenkins) Run Completed with:
> * **RESULT:** FAILURE ❌
> * **URL:** https://build.ci.opensearch.org/job/gradle-check/11984/
> * **CommitID:** [0bbd739](https://github.com/opensearch-project/OpenSearch/commit/0bbd739053cf1c3f6ac078852ab76a0192920c2a)
>   Please examine the workflow log, locate, and copy-paste the failure(s) below, then iterate to green.
>   Is the failure [a flaky test](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#flaky-tests) unrelated to your change?

@harshjain2 : Can you please rebase your branch 

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':distribution:bwc:staged:buildBwcLinuxTar'.
> Building 2.6.0 didn't generate expected file /var/jenkins/workspace/gradle-check/search/distribution/bwc/staged/build/bwc/checkout-2.6/distribution/archives/linux-tar/build/distributions/opensearch-min-2.6.0-SNAPSHOT-linux-x64.tar.gz

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
```

File: CHANGELOG.md (hunk 2 above)
Comment (harshjain2) [reply CHANGELOG.md:11]: I am not sure about it as this is my first commit. If this needs to be done, let me know. I will add it and push the commit.

Comment (harshjain2) [reply CHANGELOG.md:11]: I guess it makes sense to backport it 2.x branch since this is a bug. Pushing the commit.

Comment (reta) [inline CHANGELOG.md:11]: Please remove the entry from this section since you added it to `[Unrelease 2.x]` (it should appear only once in change log), thank you

Comment (harshjain2) [reply CHANGELOG.md]: Done.

**Review by reta — APPROVED** [review]

**Comment by Arpit-Bandejiya** [comment]
Should we add a test to check for 200 status code response for empty pit_ids as well? @reta thoughts?

**Comment by reta** [comment]
> Should we add a test to check for 200 status code response for empty pit_ids as well? @reta thoughts?

@Arpit-Bandejiya I believe `rest-api-specs` tests do check status code for success

**Comment by harshjain2** [comment]
Wht is the criteria to merge this PR. All the feedback providers must approve this PR?

**Comment by reta** [comment]
> Wht is the criteria to merge this PR. All the feedback providers must approve this PR?

Yes

**Review by dreamer-89 — APPROVED** [review]

**Comment by dreamer-89** [comment]
@reta @harshjain2 : Anything pending for merge ?

**Comment by harshjain2** [comment]
@dreamer-89 @reta I have addressed all the feedback provided on the PR. Nothing pending from my side on this PR.

**Comment by reta** [comment]
> @dreamer-89 @reta I have addressed all the feedback provided on the PR. Nothing pending from my side on this PR.

I also approved, @Arpit-Bandejiya your concerns were cleared out?

File: CHANGELOG.md [hunk 3]
```diff
@@ -73,6 +73,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - Add GeoTile and GeoHash Grid aggregations on GeoShapes. ([#5589](https://github.com/opensearch-project/OpenSearch/pull/5589))
 - Disallow multiple data paths for search nodes ([#6427](https://github.com/opensearch-project/OpenSearch/pull/6427))
 - [Segment Replication] Allocation and rebalancing based on average primary shard count per index ([#6422](https://github.com/opensearch-project/OpenSearch/pull/6422))
+- Removed NOT_FOUND for empty results. ([#6544](https://github.com/opensearch-project/OpenSearch/pull/6544))
```
Comment (andrross) [inline CHANGELOG.md:76]: Check out the [contributing doc](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#changelog) for more details, but this entry isn't super helpful as a release note because it doesn't provide any context about what this change means. I'd suggest something like "Return success on DeletePits when nothing is deleted".

File: rest-api-spec/src/main/resources/rest-api-spec/test/pit/10_basic.yml [hunk 4]
```diff
@@ -138,5 +138,7 @@
   - match: {pits.0.successful: true }
 
   - do:
-      catch: missing
-      delete_all_pits: { }
+      delete_all_pits: {}
```
Comment (andrross) [inline rest-api-spec/src/main/resources/rest-api-spec/test/pit/10_basic.yml:141]: Does this change the behavior of deleting a single PIT or a list of PITs? Should there be a test for those cases as well?

**Review by andrross — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (harshjain2) [reply CHANGELOG.md]: Done.

File: rest-api-spec/src/main/resources/rest-api-spec/test/pit/10_basic.yml (hunk 4 above)
Comment (harshjain2) [reply rest-api-spec/src/main/resources/rest-api-spec/test/pit/10_basic.yml:141]: @andrross I don't think it will result in any change in the behaviour of deleting a single PIT or a list of PITs using `DELETE localhost:9200/_search/point_in_time/` API.

### Request Delete PIT with empty array results in validation exception.
DELETE localhost:9200/_search/point_in_time/
Request Body
```
{
    "pit_id": [
        ]
}
```
Response
```
{
	"error": {
		"root_cause": [
			{
				"type": "action_request_validation_exception",
				"reason": "Validation Failed: 1: no pit ids specified;"
			}
		],
		"type": "action_request_validation_exception",
		"reason": "Validation Failed: 1: no pit ids specified;"
	},
	"status": 400
}
```

### Request with incorrect pits results in failure with failure message associated with every pit id
DELETE localhost:9200/_search/point_in_time/
Request Body
```
{
    "pit_id": [
        "o463QQEPbXktaW5kZXgtMDAwMDAxFkhGN09fMVlPUkVPLXh6MUExZ1hpaEEAFjBGbmVEZHdGU1EtaFhhUFc4ZkR5cWcAAAAAAAAAAAEWaXBPNVJtZEhTZDZXTWFFR05waXdWZwEWSEY3T18xWU9SRU8teHoxQTFnWGloQQAA",
        "o463QQEPbXktaW5kZXgtMDAwMDAxFkhGN09fMVlPUkVPLXh6MUExZ1hpaEEAFjBGbmVEZHdGU1EtaFhhUFc4ZkR5cWcAAAAAAAAAAAIWaXBPNVJtZEhTZDZXTWFFR05waXdWZwEWSEY3T18xWU9SRU8teHoxQTFnWGloQQAA"
    ]
}
```
Response
```
{
	"pits": [
		{
			"successful": false,
			"pit_id": "o463QQEPbXktaW5kZXgtMDAwMDAxFkhGN09fMVlPUkVPLXh6MUExZ1hpaEEAFjBGbmVEZHdGU1EtaFhhUFc4ZkR5cWcAAAAAAAAAAAEWaXBPNVJtZEhTZDZXTWFFR05waXdWZwEWSEY3T18xWU9SRU8teHoxQTFnWGloQQAA"
		},
		{
			"successful": false,
			"pit_id": "o463QQEPbXktaW5kZXgtMDAwMDAxFkhGN09fMVlPUkVPLXh6MUExZ1hpaEEAFjBGbmVEZHdGU1EtaFhhUFc4ZkR5cWcAAAAAAAAAAAIWaXBPNVJtZEhTZDZXTWFFR05waXdWZwEWSEY3T18xWU9SRU8teHoxQTFnWGloQQAA"
		}
	]
}
```

Comment (harshjain2) [reply rest-api-spec/src/main/resources/rest-api-spec/test/pit/10_basic.yml:141]: `Should there be a test for those cases as well?`
Few tests cases already exist for the DELETE PIT(s). But I couldn't find test case for the use case where empty pit array is passed in the body.
```
DELETE localhost:9200/_search/point_in_time/
Request Body:
{
    "pit_id": []
}
```
I believe this should be tracked through a separate issue.

**Review by andrross — APPROVED** [review]

**Comment by harshjain2** [comment]
Backport PR to 2.x : https://github.com/opensearch-project/OpenSearch/pull/6614
